executor/api/kafka/server.go

```diff
@@ -194,6 +206,17 @@ func (ks *SeldonKafkaServer) Serve() error {
 			consumerConfigMap["ssl.certificate.pem"] = sslKakfaServer.ClientCert
 		}
 		consumerConfigMap["ssl.key.password"] = sslKakfaServer.ClientKeyPass // Key password, if any
+		if util.GetKafkaSecurityProtocol() == "SASL_SSL" {                   //if we also have SASL enabled, then we need to provide the necessary settings in addition to SSL
```

Comment:
:thought_balloon: The username and password options apply for both `SASL_SSL` *and also* `SASL_PLAINTEXT`, so we should really support both here, as it's basically no extra effort.

Reply:
If we do that, this conditional should not be nested but rather another top-level conditional.